src/Apps/Order/OrderApp.tsx

```diff
@@ -99,16 +127,63 @@ export class OrderApp extends React.Component<OrderAppProps, OrderAppState> {
               />
             ) : null}
             <StripeProvider stripe={this.state.stripe}>
-              <Elements>
-                <>
-                  {children}
-                  <Spacer mb={6} />
-                </>
-              </Elements>
+              <ErrorModalContext.Provider
+                value={{ showErrorModal: this.showErrorModal }}
+              >
+                <Elements>
+                  <>
+                    {children}
+                    <Spacer mb={6} />
+                  </>
+                </Elements>
+              </ErrorModalContext.Provider>
             </StripeProvider>
+            <ErrorModal
+              show={this.state.isErrorModalOpen}
+              detailText={this.state.errorModalMessage}
+              contactEmail="orders@artsy.net"
+              headerText={this.state.errorModalTitle}
+              ctaAction={this.state.errorModalCtaAction}
+              onClose={this.hideErrorModal}
+            />
           </>
         )}
       </ContextConsumer>
     )
   }
 }
+
+export interface OrderAppPageProps {
+  showErrorModal: ShowErrorModal
+  mediator: Mediator
+}
+
+export function wrapOrderAppPage<T extends React.ComponentType>(
+  Page: T
+): T extends React.ComponentType<infer R>
+  ? React.ComponentType<
+      // strip out mediator and showErrorModal
+      { [K in Exclude<keyof R, keyof OrderAppPageProps>]: R[K] } &
+        // then add them back in as optional so they can be overridden
+        { [K in Extract<keyof R, keyof OrderAppPageProps>]?: R[K] }
+    >
+  : never {
+  return (props => (
+    <ErrorModalContext.Consumer>
+      {({ showErrorModal }) => (
+        <ContextConsumer>
+          {({ mediator }) => {
+            return (
+              // @ts-ignore weird bug saying that Page has no construct signature
+              <Page
+                showErrorModal={showErrorModal}
+                mediator={mediator}
+                {...props}
+              />
+            )
+          }}
+        </ContextConsumer>
+      )}
+    </ErrorModalContext.Consumer>
+  )) as any
```

Comment:
Could this be wrapped at the [top-most route](https://github.com/artsy/reaction/blob/master/src/Apps/Order/routes.tsx#L64) (which then composes other child routes) skipping the need to wrap each individual route?